src/main/scala/viper/gobra/translator/implementations/ContextImpl.scala

```diff
@@ -113,8 +115,18 @@ case class ContextImpl(
     pureMethodN,
     predicateN,
     builtInMembersN,
-    stmtN
+    stmtN,
+    initialFreshCounterValue = initialFreshCounterValueN
   )
 
   override def addVars(vars: LocalVarDecl*): Context = this
+
+
+  private var freshCounter: Int = initialFreshCounterValue
+  override def getFreshCounter: Int = freshCounter
+  override def getAndIncrementFreshCounter: Int = {
```

Comment:
it returns a counter that is counting fresh values and not a counter that is fresh, right? I find the name a bit confusing.

Reply:
I call it now `getAndIncrementFreshVariableCounter`. I hope this is slightly better